```{r setup, include = FALSE, message = FALSE, warning = FALSE}
library(ggplot2)
library(rnaturalearth)
library(rnaturalearthdata)
library(sf)
library(raster)
library(ggspatial)
library(viridis)
library(ggrepel)
library(tidyverse)
library(reshape2)
library(showtext)
library(sysfonts)
library(workflowr)
library(readr)
library(wesanderson)

font_add_google("Lato", "lato")

showtext_auto()
```


```{r map, message = FALSE, warning = FALSE, echo = FALSE}
world <- ne_countries(scale = "medium", returnclass = "sf")
uv_matrix <- as.matrix(read.csv("data/neo_uvi.csv", header = F))

lon <- seq(-179.75, 179.75, by = 0.5)
lat <- seq(89.75, -89.75, by = -0.5)

rownames(uv_matrix) <- lat
colnames(uv_matrix) <- lon

uv_df <- melt(uv_matrix, varnames = c("lat", "lon"), value.name = "UVIndex")

uv_df$lat <- as.numeric(as.character(uv_df$lat))
uv_df$lon <- as.numeric(as.character(uv_df$lon))
```

```{r mask, message = FALSE, warning = FALSE, echo = FALSE}
uv_raster <- rasterFromXYZ(uv_df[, c("lon", "lat", "UVIndex")], crs = "+proj=longlat +datum=WGS84")

world_sp <- as(world, "Spatial")

uv_land <- mask(uv_raster, world_sp)

uv_land_df <- as.data.frame(uv_land, xy = TRUE)
names(uv_land_df) <- c("lon", "lat", "UVIndex")
```

```{r sites, echo = FALSE}
sites <- data.frame(
  name = c("Denisova","Chagyrskaya","Vindija", "Mezmaiskaya", 
           "El Sidron", "Spy","Les Cottés"),
  samples = c("Denisova 3 (Denisovan)\nDenisova 11 (F1 Hybrid)\n Denisova 5 (Neanderthal)\n Denisova 25 (Denisovan)",
    "Chagyrskaya 8 (Neanderthal)",
    "Vindija 33.19 (Neanderthal)",
    "Mezmaiskaya 1 (Neanderthal)\n Mezmaiskaya 2 (Neanderthal)", 
    "Sid1253 (Neanderthal)",
    "Spy 94-a (Neanderthal)",
    "Les Cottés Z4-1415 (Neanderthal)"),
  lon  = c(84.7, 83.1, 16.8, 40, -5.3, 4.7, 2.5), #E (+) W (-)
  lat  = c(51.4, 51.3, 46.3, 44.1, 43.4, 50.5, 46.2) #N (+) S (-)
)
sites_sf <- st_as_sf(sites, coords = c("lon","lat"), crs = 4326)
```


# Genome availability

```{r plot, echo = FALSE}
wes_cols <- wes_palette("Zissou1", 100, type = "continuous")

ggplot() +
  geom_raster(data = uv_land_df, aes(x = lon, y = lat, fill = UVIndex)) +
  scale_fill_gradientn(colours = wes_cols, name = "UV Index", na.value = "white") +
  geom_sf(data = world, fill = NA, color = "grey60", size = 0.2) +
  geom_point(data = sites, aes(x = lon, y = lat), 
             color = "#00008B", fill = "#00008B", shape = 21, size = 3, stroke = 1) +
  geom_label_repel(
    data = sites,
    aes(x = lon, y = lat, label = samples),
    box.padding = 0.7,
    point.padding = 0.3,
    segment.color = "#00008B",
    segment.size = 0.4,
    fill = alpha("white", 0.8),
    label.r = unit(0.2, "lines"),
    color = "black",
    size = 3,
    label.size = 0.3,
    family = "lato",
    min.segment.length = 0
  ) +
  coord_sf(expand = FALSE, xlim = c(-20, 150), ylim = c(20, 70)) +
  theme_minimal(base_family = "lato") +
  theme(
    legend.position = c(0.9, 0.3),
    legend.background = element_blank(),
    panel.grid = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.title = element_blank(),
    plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
    plot.subtitle = element_text(size = 10, hjust = 0.5)
  ) +
  labs(
    title = "Ancient DNA Discovery Sites and UV Radiation",
    subtitle = "UV Index climatology (NASA AURA, 2010)"
  )

```

# Modern reference genomes

```{r modern_samples, echo = FALSE}
modern_samples <- data.frame(
  population = c("Yoruba", "Bantu_Hetero", "Khomani_San", "Mongola", "Tlingit",
                 "Eskimo_Sireniki", "Dai", "Cambodian", "Han", "Papuan", "Maori",
                 "Australian", "French", "Hungarian", "Greek"),
  lon  = c(3.9, 19.0, 20.8, 111.0, 158.65, 173.9, 100.0, 105.0, 114.0, 143.0, 
           174.5, 143.0, 2.0, 9.1, 23.7), #E (+) W (-)
  lat  = c(7.4, -22.0, -27.0, 45.0, 53, 64.4, 21.0, 12.0, 32.3, -4.0, -41.3,
           -13.0, 46.0, 47.5, 38.0) #N (+) S (-)
)

modern_sf <- st_as_sf(sites, coords = c("lon","lat"), crs = 4326)

ggplot() +
  geom_raster(data = uv_land_df, aes(x = lon, y = lat, fill = UVIndex)) +
  scale_fill_gradientn(colours = wes_cols, name = "UV Index", na.value = "white") +
  geom_sf(data = world, fill = NA, color = "grey60", size = 0.2) +
  geom_point(data = modern_samples, aes(x = lon, y = lat), 
             color = "#00008B", fill = "#00008B", shape = 21, size = 3, stroke = 1) +
  geom_label_repel(
    data = modern_samples,
    aes(x = lon, y = lat, label = population),
    box.padding = 0.7,
    point.padding = 0.3,
    segment.color = "#00008B",
    segment.size = 0.4,
    fill = alpha("white", 0.8),
    label.r = unit(0.2, "lines"),
    color = "black",
    size = 3,
    label.size = 0.3,
    family = "lato",
    min.segment.length = 0
  ) +
  coord_sf(expand = FALSE, xlim = c(-20, 200), ylim = c(-50, 80)) +
  theme_minimal(base_family = "lato") +
  theme(
    legend.position = c(0.9, 0.3),
    legend.background = element_blank(),
    panel.grid = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.title = element_blank(),
    plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
    plot.subtitle = element_text(size = 10, hjust = 0.5)
  ) +
  labs(
    title = "Modern Samples and UV Radiation",
    subtitle = "UV Index climatology (NASA AURA, 2010)"
  )
```



# Distribution of pigmentation snps

```{r snps_dist, echo = FALSE, message = FALSE, warning = FALSE,}
snps_set <- read.csv("data/pigmentation_snps.csv", header = T)

snps_set <- snps_set %>%
  mutate(Chromosome = sub("\\:.*", "", locations),
         Chromosome = factor(Chromosome, 
                             levels = sort(as.numeric(unique(Chromosome)))
                             )
         )


ggplot(snps_set, aes(x = Chromosome)) +
  geom_bar(
    fill = wes_palette("Zissou1", 1),
    color = "black",
    linewidth = 0.3,
    width = 0.8
  ) +
  labs(
    x = "Chromosome",
    y = "Number of SNPs"
  ) +
  theme_classic(base_family = "lato", base_size = 14) +
  theme(
    axis.title = element_text(face = "bold"),
    axis.text = element_text(color = "black"),
    axis.line = element_line(color = "black", linewidth = 0.4),
    panel.grid.major.y = element_line(
      color = "grey85",
      linewidth = 0.3
    ),
    panel.grid.minor = element_blank()
  )
```

# Sequencing coverage

```{r depth, include = FALSE, message = FALSE, warning = FALSE, echo = FALSE}
scladina <- read.table("data/Sclad_depth.txt", header = F) 
denny <- read.table("data/denny_depth.txt", header = F) 
mez1 <- read.table("data/mez1_depth.txt", header = F) 
hstadel <- read.table("data/hst_depth.txt", header = F) 
spy <- read.table("data/spy_depth.txt", header = F) 
lescottes <- read.table("data/lesCottes_depth.txt", header = F)
den5 <- read.table("data/den5_depth.txt", header = F)
mez2 <- read.table("data/mez2_depth.txt", header = F)
vindija87 <- read.table("data/vindija87_depth.txt", header = F)
chag8 <- read.table("data/Chagyrskaya8_depth.txt", header = F)
den25 <- read.table("data/den25_depth.txt", header = F)

den3 <- read.table("data/Denisova3_depth.txt", header = F)

colnames(mez2) <- c("Chromosome", "Position", "Depth")
colnames(lescottes) <- c("Chromosome", "Position", "Depth")
colnames(hstadel) <- c("Chromosome", "Position", "Depth")
colnames(scladina) <- c("Chromosome", "Position", "Depth")
colnames(denny) <- c("Chromosome", "Position", "Depth")
colnames(mez1) <- c("Chromosome", "Position", "Depth")
colnames(spy) <- c("Chromosome", "Position", "Depth")
colnames(den5) <- c("Chromosome", "Position", "Depth")
colnames(chag8) <- c("Chromosome", "Position", "Depth")
colnames(vindija87) <- c("Chromosome", "Position", "Depth")
colnames(den25) <- c("Chromosome", "Position", "Depth")
colnames(den3) <- c("Chromosome", "Position", "Depth")
den3 <- den3 %>%
  mutate(Chromosome = as.numeric(substr(Chromosome, 3, nchar(Chromosome))))

scladina$Sample <- "Scladina"
denny$Sample <- "Denisova 11 (Denny)"
mez1$Sample <- "Mez1"
mez2$Sample <- "Mez2"
hstadel$Sample <- "HST"
spy$Sample <- "Spy"
lescottes$Sample <- "les Cottes"
den5$Sample <- "Denisova 5 (Altai Neanderthal)"
chag8$Sample <- "Chagyrskaya 8"
vindija87$Sample <- "Vindija 87"
den25$Sample <- "Denisova 25"
den3$Sample <- "Denisova 3"

depth_all <- bind_rows(scladina, denny, mez1, mez2, hstadel, 
                       spy, lescottes, den5, chag8, vindija87, 
                       den25, den3)

depth_all <- depth_all %>%
  mutate(
    Chromosome = factor(Chromosome, levels = unique(Chromosome)),
    Sample = factor(Sample, levels = c("Scladina", "Denisova 11 (Denny)", "Mez1", "Mez2", "HST", "Spy", "les Cottes", "Denisova 5 (Altai Neanderthal)", "Chagyrskaya 8", "Vindija 87", "Denisova 25", "Denisova 3"))
  )
```

```{r depth_viz, echo = FALSE}
wes_samples <- wes_palette("Zissou1", 12, type = "continuous")

ggplot(depth_all, aes(x = Position, y = Depth, color = Sample)) +
  geom_point(size = 2, alpha = 0.6) +
  scale_color_manual(values = wes_samples) +
  labs(
    title = "Sequencing Depth",
    subtitle = "EBI pigmentation-related loci",
    x = "Genomic Position",
    y = "Sequencing depth",
    color = "Sample"
  ) +
  theme_bw(base_size = 14) +
  theme_minimal(base_family = "lato") +
  theme(
    legend.position = "top",
    legend.background = element_blank(),
    panel.grid = element_blank(),
    plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
    plot.subtitle = element_text(size = 10, hjust = 0.5)
  )
```


```{r depth_viz_log, echo = FALSE}
ggplot(depth_all, aes(x = Position, y = log(Depth), color = Sample)) +
  geom_point(size = 2, alpha = 0.6) +
  scale_color_manual(values = wes_samples) +
  labs(
    title = "Sequencing Depth (log-transformed)",
    subtitle = "EBI pigmentation-related loci",
    x = "Genomic Position",
    y = "Sequencing depth (log)",
    color = "Sample"
  ) +
  theme_bw(base_size = 14) +
  theme_minimal(base_family = "lato") +
  theme(
    legend.position = "top",
    legend.background = element_blank(),
    panel.grid = element_blank(),
    axis.text.x = element_blank(),
    plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
    plot.subtitle = element_text(size = 10, hjust = 0.5)
  )
```





